### PR TITLE
Change migration to loop through formulae.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

The regex in `migrate_cache_entries_to_double_dashes` will never catch all edge cases.

### CODE REVIEW

File: Library/Homebrew/compat/download_strategy.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class AbstractFileDownloadStrategy
+  # TODO: This can be removed after a month because downloads
```
Comment (MikeMcQuaid) [inline Library/Homebrew/compat/download_strategy.rb]: I don't agree with this; I'd rather we kept around something indefinitely as there's plenty of things that get updated less than once a month. Can the same logic not live in `brew update` indefinitely?

**Comment by reitermarkus** [comment]
Another solution would be to loop through all formulae in the migration and to the same thing the `Compat` module does.

**Comment by MikeMcQuaid** [comment]
> Another solution would be to loop through all formulae in the migration and to the same thing the Compat module does. This way it doesn't rely on a regex.

@reitermarkus Sounds good to me as long as the no-op case is fast 👍

**Comment by reitermarkus** [comment]
> Sounds good to me as long as the no-op case is fast

Not sure about the performance, but it will be a no-op when updating from `> 1.7.1`.

**Comment by MikeMcQuaid** [comment]
@reitermarkus Sounds good.

**Pull request merged**